### PR TITLE
🔧 Make mypy and doc string linters opt out instead of opt in

### LINKED ISSUE #1001: 🔧 Make mypy and doc string linters opt out instead of opt in
### Proposed refactoring

To improve our code quality we started to use `mypy` and docstring linters on more and more sub packages.
We are now at a point where it might be easier to those tools opt out for the packages that aren't ready, instead of opt in for the packages that we already use them on.

Currently checked sub packages:
- plugin_system
- utils
- deprecation
- testing
- parameter
- project
- model/property.py
- builtin/io/pandas

Currently **NOT** checked sub packages:
- io (should be no problem)
- typing (should be no problem)
- analysis (needs type stubs and docstrings)
- builtin (needs type stubs  and docstrings for megacomplexes, io needs more annotations and docstrings)
- cli (?)
- model (needs type stubs and docstrings)

### Benefits

- [x] Makes code easier to maintain
- [x] Documents the user facing API
- [x] Mypy will find a lot of problem we didn't discover yet (unused code paths)

### PR DESCRIPTION
**Pull request by s-weigand**
Changed the `mypy` and `flake8-docs` pre-commit hooks to be opt out instead of opt in and fixed some docstring and typing issues along the way.

### Change summary

- [🔧⌨️ Made mypy optout instead of opt in and fixed some typing issues](https://github.com/glotaran/pyglotaran/commit/aeb4de0db06226c365c5f73adfdac1bcd94a8b61)
- [🔧👌 Moved mypy config from setup.cfg to pyproject.toml](https://github.com/glotaran/pyglotaran/commit/1935679ceeac4af2d270bc4974848726180511ae)
- [🔧👌 Made docstring linter opt out instead of opt in](https://github.com/glotaran/pyglotaran/commit/f1560a35f6cb0e18734ba982136c1a73d0148bd3)
- [🩹 Fix yesqa by moving count from the config to the hook args](https://github.com/glotaran/pyglotaran/commit/ec502da984b12668d4bbff17f232b1cbfd27462b)

### Checklist

- [x] ✔️ Passing the tests (mandatory for all PR's)
- [x] 🚧 Added changes to changelog (mandatory for all PR's)
- [x] 👌 Closes issue (mandatory for ✨ feature and 🩹 bug fix PR's)

### Closes issues

closes #1001

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -29,6 +29,8 @@ per-file-ignores =
     docs/source/conf.py: E501
     # Typedef files are formatted differently
     *.pyi: E301, E302, F401
+    # D414 is needed due to a bug in pydocstyle that sees the returntype `Parameters` as nested sections
```
Comment (jsnel) [inline tox.ini:32]: 'ReturnType' in the context of Javascript, 'return type' otherwise.

File: glotaran/builtin/io/netCDF/netCDF.py [hunk 2]
```diff
@@ -22,6 +36,16 @@ def save_dataset(
...
+        """Write a :xarraydoc:`Dataset` to the ``*.nc`` at path ``file_name``.
 
+        Parameters
+        ----------
+        dataset: xr.Dataset
+            :xarraydoc:`Dataset` that should be written to file.
+        file_name: str
+            Path of the file to write ``dataset`` to.
+        data_filters: list[str] | None
+            List of data variable names that should be written to file. Defaults to None
```
Comment (jsnel) [inline glotaran/builtin/io/netCDF/netCDF.py:48]:
```suggestion
            List of data variable names that should be written to file. Defaults to None.
```

File: glotaran/builtin/io/sdt/sdt_file_reader.py [hunk 3]
```diff
@@ -26,39 +26,41 @@ def load_dataset(
...
-        index: list, np.ndarray
+        index: np.ndarray | None
             This is only needed if `type_of_data=="st"`, since `*.sdt` files,
             which only contain spectral temporal data, lack the spectral information.
             Thus for the spectral axis data need to be given by the user.
 
-        flim:
-            Set true if reading a result from a FLIM measurement.
+        flim: bool
+            Set true if reading a result from a FLIM measurement. Defaults to False
```
Comment (jsnel) [inline glotaran/builtin/io/sdt/sdt_file_reader.py:42]:
```suggestion
            Set true if reading a result from a FLIM measurement. Defaults to False.
```

File: glotaran/builtin/io/sdt/sdt_file_reader.py [hunk 4]
```diff
@@ -26,39 +26,41 @@ def load_dataset(
...
 
-        dataset_index: int: default 0
+        dataset_index: int
             If the `*.sdt` file contains multiple datasets the index will used
-            to select the wanted one
+            to select the wanted one. Defaults to 0
```
Comment (jsnel) [inline glotaran/builtin/io/sdt/sdt_file_reader.py:46]:
```suggestion
            to select the wanted one. Defaults to 0.
```

File: glotaran/builtin/io/sdt/sdt_file_reader.py [hunk 5]
```diff
@@ -26,39 +26,41 @@ def load_dataset(
...
 
-        swap_axis: bool, default False
+        swap_axis: bool
             Flag to switch a wavelength explicit `input_df` to time explicit `input_df`,
-            before generating the SpectralTemporalDataset.
+            before generating the SpectralTemporalDataset. Defaults to False
```
Comment (jsnel) [inline glotaran/builtin/io/sdt/sdt_file_reader.py:50]:
```suggestion
            before generating the SpectralTemporalDataset. Defaults to False.
```

File: glotaran/builtin/io/yml/utils.py [hunk 6]
```diff
@@ -1,4 +1,4 @@
-"""Utility functionality module for ``glotaran.builtin.io.yml.yml``"""
+"""Utility functionality module for ``glotaran.builtin.io.yml.yml``."""
```
Comment (jsnel) [inline glotaran/builtin/io/yml/utils.py:1]:
```suggestion
"""Utility module for ``glotaran.builtin.io.yml.yml``."""
```
'Functionality' doesn't add anything here, just 'utility module' or "Yaml utility module".

**Review by jsnel — CHANGES_REQUESTED** [review]
Small stylistic improvements.

File: tox.ini (hunk 1 above)
Comment (s-weigand) [reply tox.ini:32]:
```suggestion
    # D414 is needed due to a bug in pydocstyle that sees the return type `Parameters` as nested sections
```

**Pull request merged**